internal/cli/atlas/accesslists/create.go

```diff
@@ -80,26 +83,85 @@ func (opts *CreateOpts) newProjectIPAccessList() []*atlas.ProjectIPAccessList {
 	return []*atlas.ProjectIPAccessList{entry}
 }
 
+func iPAddress() (string, error) {
+	if publicIP := store.IPAddress(); publicIP != "" {
+		return publicIP, nil
+	}
+
+	return "", errors.New("not able to find your public IP address. Please providing the desired IP address for this command")
+}
+
+func (opts *CreateOpts) needsArg() bool {
+	// Unless currentIP flag is enabled and type is ip address, args are required.
+	return !(opts.entryType == ipAddress && opts.currentIP)
+}
+
+func (opts *CreateOpts) validateCurrentIPFlag(cmd *cobra.Command, args []string) func() error {
+	return func() error {
+		if !opts.needsArg() && len(args) > 0 {
+			return fmt.Errorf(
+				"please either provide [entry] or use %s to use current IP Address.\n\nUsage: %s",
+				flag.CurrentIP,
+				cmd.UseLine(),
+			)
+		}
+
+		if opts.needsArg() && len(args) == 0 {
+			return fmt.Errorf(
+				"%q with entry type %s requires at least %d argument received %d\n\nUsage:  %s",
+				cmd.CommandPath(),
+				opts.entryType,
+				1,
+				len(args),
+				cmd.UseLine(),
+			)
+		}
+
+		if opts.needsArg() && opts.currentIP {
+			return fmt.Errorf("%q with entry type %s does not support %s flag.\n\n Usage: %s",
+				cmd.CommandPath(),
+				opts.entryType,
+				flag.CurrentIP,
+				cmd.UseLine(),
+			)
+		}
```

Comment:
[nit] this is a good effort but, 1) the two first checks are the same, 2) the third one could be better, not to mention the help may be confusing here as `cmd.UseLine(),` won't print flag usage

A simpler way to do this, 
for the two first: `if len(args) !=0 && opts.currenIP { fmt.Errorf("entry %s and currentIp are mutually exclusive")`
for the second: `if opts.entryType != ipAddress && opts.currenIP { fmt.Errorf("type and currentIp are mutually exclusive")`

Reply:
Please note I'm keeping the code simple on the above example just a general idea

Reply:
cool, that simplifies! I was trying to give a personalized output to each situation but I think you summarized well above :D will do.

Reply:
but actually we're not covering empty entry in this case, we still need 3 `if/else `conditions

Reply:
If you agree, I'd like to keep this as is.

Reply:
what empty string? none of your check are for empty string

Reply:
Sorry, I meant empty args. ` len(args) == 0`

Reply:
let's keep it, I left a couple fo comments on fixing the formatting and maybe simplifying one of the `if`s conditions